-1

**Failed Tests**:   Build 
**Size**: This PR adds an extra 20KB to repository
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-5093bb/47957/summary.html
**COMMIT**: 1d7277465d33c14493e8e911c7ddd2417e7b215d
**CMSSW**: CMSSW_15_1_X_2025-09-03-1400/el8_amd64_gcc12
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/48839/47957/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Build

I found compilation error when building: 

<pre>/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02905/el8_amd64_gcc12/external/gcc/12.3.1-40d504be6370b5a30e3947a6e575ca28/bin/../lib/gcc/x86_64-redhat-linux-gnu/12.3.1/../../../../x86_64-redhat-linux-gnu/bin/ld.bfd: tmp/el8_amd64_gcc12/src/PerfTools/AllocMonitorPreload/src/PerfToolsAllocMonitorPreload/ccQMU2oa.ltrans0.ltrans.o: in function `aligned_alloc':
<artificial>:(.text+0x877): undefined reference to `dlsym'
/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02905/el8_amd64_gcc12/external/gcc/12.3.1-40d504be6370b5a30e3947a6e575ca28/bin/../lib/gcc/x86_64-redhat-linux-gnu/12.3.1/../../../../x86_64-redhat-linux-gnu/bin/ld.bfd: tmp/el8_amd64_gcc12/src/PerfTools/AllocMonitorPreload/src/PerfToolsAllocMonitorPreload/ccQMU2oa.ltrans0.ltrans.o: in function `posix_memalign':
<artificial>:(.text+0x942): undefined reference to `dlsym'
/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02905/el8_amd64_gcc12/external/gcc/12.3.1-40d504be6370b5a30e3947a6e575ca28/bin/../lib/gcc/x86_64-redhat-linux-gnu/12.3.1/../../../../x86_64-redhat-linux-gnu/bin/ld.bfd: tmp/el8_amd64_gcc12/src/PerfTools/AllocMonitorPreload/src/PerfToolsAllocMonitorPreload/ccQMU2oa.ltrans0.ltrans.o:<artificial>:(.text+0xaa7): more undefined references to `dlsym' follow
collect2: error: ld returned 1 exit status
gmake: *** [tmp/el8_amd64_gcc12/src/PerfTools/AllocMonitorPreload/src/PerfToolsAllocMonitorPreload/libPerfToolsAllocMonitorPreload.so] Error 1
Leaving library rule at PerfTools/AllocMonitorPreload
>> Leaving Package PerfTools/AllocMonitorPreload
>> Package PerfTools/AllocMonitorPreload built
>> Entering Package PerfTools/AllocMonitor
</pre>

--- a/PerfTools/AllocMonitor/BuildFile.xml
+++ b/PerfTools/AllocMonitor/BuildFile.xml
@@ -1,3 +1,7 @@
-<export>
-  <lib name="1"/>
-</export>
+<environment>
+  <use name="sockets"/> <!-- this provides the dl library -->
+  <use name="FWCore/Utilities" source_only="1"/>
+  <export>
+    <lib name="1"/>
+  </export>
+</environment>

